### PR TITLE
feat(api): enable cors on codegen asset bucket

### PR DESCRIPTION
**Pull request by sundersc**
#### Description of changes

Enable CORS on the codegen assets bucket. This is required for console team to access the objects from user context.

##### CDK / CloudFormation Parameters Changed

No.

#### Issue #, if available
NA.

#### Description of how you validated changes
- Local Testing
- Added E2E tests

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-core/src/utils/sdk-calls.ts [hunk 1]
```diff
@@ -205,3 +205,33 @@ export const listAttachedRolePolicies = async (roleName: string, region: string)
...
+  return null;
+};
+
+export const getBucketCorsPolicy = async (bucketName: string, region: string): Promise<Record<string, any>[]> => {
+  const service = new S3({ region });
+  const params = {
+    Bucket: bucketName,
+  };
+  const corsPolicy = await service.getBucketCors(params).promise();
+  return corsPolicy.CORSRules;
```
Comment (dzhan-aws) [inline packages/amplify-e2e-core/src/utils/sdk-calls.ts:236]: According to [the documentation](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#getBucketCors-property), `corsPolicy.CORSRules` would be an array of records, which wouldn't match the return type in the signature.

Comment (sundersc) [reply packages/amplify-e2e-core/src/utils/sdk-calls.ts:236]: Yes, the return type I have is an array of `Record<string, any>`.

File: packages/amplify-e2e-core/src/utils/sdk-calls.ts [hunk 2]
```diff
@@ -205,3 +205,33 @@ export const listAttachedRolePolicies = async (roleName: string, region: string)
   const service = new IAM({ region });
   return (await service.listAttachedRolePolicies({ RoleName: roleName }).promise()).AttachedPolicies;
 };
+
+export const getBucketNameFromModelSchemaS3Uri = (uri: string | null): string => {
```
[the remark this answers is not in the document]
Comment (sundersc) [reply packages/amplify-e2e-core/src/utils/sdk-calls.ts:209]: Updated.

**Review by dzhan-aws — APPROVED** [review]

**Review by phani-srikar — APPROVED** [review]

**Review by dpilch — DISMISSED** [review]

**Review by phani-srikar — DISMISSED** [review]

**Review by palpatim — APPROVED** [review]

**Comment by palpatim** [comment]
Converting this to draft pending further review

**Review by palpatim — APPROVED** [review]

**Review by dpilch — APPROVED** [review]

**Pull request merged**